src/Writer/CsvWriter.php

```diff
@@ -126,6 +126,12 @@ public function write(array $data): void
             ++$this->position;
         }
 
+        if (1 !== \strlen($this->delimiter) ||
+            1 !== \strlen($this->enclosure) ||
+            1 !== \strlen($this->escape)) {
+            throw new InvalidDataFormatException();
```

Comment:
Please always provide a message when throwing.
Here is a suggestion:
```suggestion
            throw new InvalidDataFormatException(<<<'EXCEPTION'
            Context: trying to write CSV data
            Problem: delimiter, enclosure or escape character is actually a
            string longer than just one character
            Solution: pick an actual character
            EXCEPTION
            );
```

Even then, it could be improved by showing exactly which character is not just a character.

Reply:
Does this works for php 7.3 ?

Reply:
[Yes](https://3v4l.org/VG0k0)